site/docs/concepts/optimizer.md

```diff
@@ -0,0 +1,26 @@
+# Optimizer
+
+The optimizer component builds an optimal data-plane plotter for a given `FybrikApplication` custom resource, taking into account:
+* Available resources (modules, clusters, storage accounts)
+* Application specification (e.g., geography, data access protocol)
+* Specifications of the data-sets required by the application
+* Governance actions required by the data-governance policy manager
+* [Configuration policies](./config-policies)
+* Optimization goals
+
+The optimizer translates all the above data into a monolith [Constraint Satisfaction Problem (CSP)](https://en.wikipedia.org/wiki/Constraint_satisfaction_problem) and solves it using a third-party CSP solver. The solver returns an optimal solution in terms of the specified optimization goals. The solution is then translated into a data-plane plotter. The plotter specifies which modules should be deployed in which clusters, using which storage accounts and which configuration. Finally, the plotter is deployed to the specified clusters, resulting in a data plane that connects the required data-sets to the application.
+
+## Using a custom CSP solver
```

Comment:
I would suggest putting this section as a new page under https://fybrik.io/v1.0.0/tasks rather than here